llvm/lib/Target/AMDGPU/Utils/AMDGPUBaseInfo.cpp

```diff
@@ -1449,6 +1449,15 @@ unsigned getMaxNumVGPRs(const MCSubtargetInfo *STI, unsigned WavesPerEU,
   return std::min(MaxNumVGPRs, AddressableNumVGPRs);
 }
 
+unsigned getMaxNumAGPRs(const MCSubtargetInfo *STI, unsigned int WavesPerEU) {
+  assert(WavesPerEU != 0);
+
+  unsigned MaxNumAGPRs = getTotalNumAGPRs() / WavesPerEU;
```

Comment:
I'm not sure we should be using 256 for the numerator. For example, we can have 256 AGPRs on MI300 with 2-waves-per-eu (total VGPR + AGPR must be <= 256, for 1 wave that number is 512, but were capped to 256 AGPR because that is the total number provided by hardware)